File: roll_cant_stop.py
from random import randint
from typing import List, Tuple
from sys import argv


class Die():
    def __init__(self, size=6):
        self.current_value = None
        self._size = size

    def roll(self):
        self.current_value = randint(1, self._size)


def gen_pairs(dice_results: List[Die]) -> List[Tuple[int]]:
    pairs = []
    pairs_of_pairs = []
    number_of_dice_rolls = len(dice_results)
    for _ in range(0, number_of_dice_rolls):
        current_el = dice_results.pop(0)
        for j in dice_results:
                pairs.append((current_el,j))

    number_of_pairs = len(pairs)

    for _ in range(0, number_of_pairs):
        current_el = pairs.pop(0)
        for j in pairs:
            if ((current_el[0] not in j) & (current_el[1] not in j)):
                pair0_value0 = current_el[0].current_value
                pair0_value1 = current_el[1].current_value
                pair1_value0 = j[0].current_value
                pair1_value1 = j[1].current_value
                pairs_of_pairs.append(((pair0_value0, pair0_value1),(pair1_value0, pair1_value1)))

    return pairs_of_pairs

def roll_dice(number_of_dice:int, size_of_dice:int) -> List[Die]:
    dice_rolls = []
    for _ in range(0, number_of_dice):
        die = Die(size=size_of_dice)
        die.roll()
        dice_rolls.append(die)

    return dice_rolls



if __name__ == '__main__':
    dice_rolls = roll_dice(int(argv[1]), int(argv[2]))
    pairs_of_pairs = gen_pairs(dice_rolls)
    print(pairs_of_pairs)

File: roll_cant_stop.js
class Die {
    constructor(size=6) {
        this.current_value = null;
        this.size = size;
    }

    roll() {
        this.current_value = randInt(this.size);
    }
}

let randInt = (max, min=1) => {
    return Math.floor(Math.random() * (max - min + 1) + min);
}

let gen_pairs = (dice_results) => {
    let pairs = [];
    let pairs_of_pairs = [];
    let number_of_dice_rolls = dice_results.length;
    for (let i = 0; i < number_of_dice_rolls; i++) {
        let current_el = dice_results.pop(0);
        dice_results.forEach((j) => {
            pairs.push([current_el, j]);
        })
    }

    let number_of_pairs = pairs.length
    for (let i = 0; i < number_of_pairs; i++) {
        let current_el = pairs.pop(0)
        pairs.forEach((j) => {
            if(!(j.includes(current_el[0])) & !(j.includes(current_el[1]))){
                let pair0_value0 = current_el[0].current_value
                let pair0_value1 = current_el[1].current_value
                let pair1_value0 = j[0].current_value
                let pair1_value1 = j[1].current_value
                pairs_of_pairs.push([[pair0_value0, pair0_value1],[pair1_value0, pair1_value1]])                
            }
        })
    }

    return pairs_of_pairs
}

let roll_dice = (numberOfDice, sizeOfDice) => {
    let dice_rolls = [];
    let i = 0;
    do {
        die = new Die(size=sizeOfDice);
        die.roll();
        dice_rolls.push(die);
        i += 1;

    } while (i < numberOfDice);

    return dice_rolls
}


const args = process.argv.slice(2);
console.log(gen_pairs(roll_dice(args[0], args[1])))